### PR TITLE
no-color => color option rename

### PR DESCRIPTION
**Pull request by tombruijn**
Commands like `bundle show` offer a `--no-color` option, but also create a `--no-no-color` option, which is a bit odd.

```
$ bundle help show
Usage:
  bundle show GEM [OPTIONS]

Options:
      [--no-color], [--no-no-color]         # Disable colorization in output
```

This change changes the option to be `color` first and `no-color` second. It still supports `--no-color`, as Thor provides this option by default for boolean options, and prevents a `--no-no-color` option from being created using the `--color` option instead.

```
$ dbundle help show
Usage:
  bundle show GEM [OPTIONS]

Options:
      [--color], [--no-color]                   # Colorize output
```

I hope the tests all pass, because I got many random failing specs locally. I will try to fix the specs if they do occur, but how can I get it to run stable local as well? Running it in a clean vagrant machine on ruby 2.3. (edit update: seemed to work fine on travis)

### CODE REVIEW

**Comment by segiddins** [comment]
👍🏻

**Comment by RochesterinNYC** [comment]
This is great! "No No Color", hahah.

**Comment by simi** [comment]
1. This will probably need deprecation cycle or target 2.0 version / branch.
2. There are also `no-cache`, `no-prune`, `no-install` options facing same problem.

**Comment by tombruijn** [comment]
Oh, I didn't know about the other options, only ran into `no-no-color` myself.

What would be the recommended approach of deprecation? Option alias in Thor or adding another option that sets the same value internally?

**Comment by segiddins** [comment]
@simi why do you think this is a breaking change?

**Comment by RochesterinNYC** [comment]
I can handle the changes for the other `--no` options, unless you want to keep working on these @tombruijn ?

**Comment by simi** [comment]
If I understand well, this removes `--no-no-color` option.

**Comment by tombruijn** [comment]
@RochesterinNYC I'm already updating it :smile:  I'll update the PR tomorrow

**Comment by segiddins** [comment]
@simi that option has never been documented, afaict, and was an accidental side-effect.

**Comment by simi** [comment]
I understand, but it is part of the help message.

```
bundle --help show
Usage:
  bundle show GEM [OPTIONS]

Options:
      [--paths=List the paths of all gems that are required by your Gemfile.], [--no-paths]   
      [--outdated=Show verbose output including whether gems are outdated.], [--no-outdated]  
      [--no-color], [--no-no-color]                                                           # Disable colorization in output
  -r, [--retry=NUM]                                                                           # Specify the number of times you wish to attempt network commands
  -V, [--verbose], [--no-verbose]                                                             # Enable verbose output mode
```

**Comment by segiddins** [comment]
Meh. I personally wouldn't classify that as a breaking change, since we've never intended it to be that way.

File: lib/bundler/cli/cache.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def run
       setup_cache_all
       Bundler.settings[:cache_all_platforms] = options["all-platforms"] if options.key?("all-platforms")
       Bundler.load.cache
-      Bundler.settings[:no_prune] = true if options["no-prune"]
```
Comment (segiddins) [inline lib/bundler/cli/cache.rb]: these settings changes are however backwards incompatible

**Comment by tombruijn** [comment]
@segiddins Yes, sorry about that. I'm aware this was backwards incompatible and my idea was to add support for this and some deprecation warnings. However, I cannot seem to get the test suite running locally properly, that's why I've been pushing to travis to test it.

However, to now continue to add the backwards compatibility with only Travis as a test environment is very inefficient. I'm pushing a backwards compatible version that does not change the settings, is only about `no-color` and `no-prune` and should be okay to merge. I'm afraid someone else will have to update the other options.

I just have one question about deprecation. Because of the rename from `no-color` to `color` the fetch `[]` call from the options hash also changes. Is this okay? No other gems/plugins/extensions depend on bundler still using the `options["no-color"]` option? This will now always return `nil` and only the `color` key will return the option value.

---

About the test suite, what can I do to get it working properly? I'll link my setup https://gist.github.com/tombruijn/6d2994b287b43eb4dd59 a Vagrantfile and the script I use to run it.

When I run the test suite on the master branch I get this result: https://gist.github.com/tombruijn/566fab7d9b4d8af7645a

File: lib/bundler/cli/cache.rb (hunk 1 above)
Comment (segiddins) [inline lib/bundler/cli/cache.rb]: this will break if `--no-prune` is used and the current setting is `true`?

**Comment by indirect** [comment]
That kind of internals change seems okay to me. For the test suite, I'm honestly not sure what to tell you. The tests pass on Travis' Ubuntu 12.04 machines, which seem like they should be the same as your test environment. You could try installing Ruby some way other than RVM, or using a different base Linux? I don't know. :(

**Comment by tombruijn** [comment]
@indirect thanks, if that had to be supported as well it would have been quite the change

About the test suite, it's getting better. I reliably only have two errors now if I don't run it in a vagrant shared directory :sweat_smile:

File: lib/bundler/cli/cache.rb (hunk 1 above)
Comment (tombruijn) [inline lib/bundler/cli/cache.rb]: @segiddins the easiest way would be `Bundler.settings[:no_prune] = !options["prune"]`

or do I need to consider that this option can be set by something else (like a bundler config file)? If that's the case I need to explicitly check if the option is given to set it. If it is given it sets it to the `!options["prune"]` value. If it's not given it defaults to `true` if the settings is not set already

**Comment by segiddins** [comment]
@indirect thoughts on merging this PR?

**Comment by tombruijn** [comment]
@segiddins  Sorry for the lack of updates. I'm resubmitting this PR with only the color flag update, it's what I had originally. Couldn't get the changes for the `--no-no-` flags right. I hope someone else can update them :)

**Comment by segiddins** [comment]
@tombruijn no worries -- its actually mergable as far as I can see?

**Comment by tombruijn** [comment]
Yep, should be. Tests are green and flag still works as intended

**Comment by segiddins** [comment]
@indirect please review

**Comment by segiddins** [comment]
@homu delegate=RochesterinNYC

**Comment by tombruijn** [comment]
Rebased and updated, but I'm not sure what I need to do to get appveyor to work?

**Comment by simi** [comment]
> — AppVeyor was unable to build non-mergeable pull request

You need to rebase against **latest** master.

**Comment by tombruijn** [comment]
Updated again :)
Let's see what Travis says

**Comment by tombruijn** [comment]
I'm abandoning this PR. Having tried to resolve these new issues that popped up after rebasing I realized I no longer want to continue working on it.

I'm pushing my original changes, which were green about three months ago, and hope someone with more experience with this code base wants to tackle this issue.

**Pull request closed without merging**